### PR TITLE
Confidential VM in Vmss support

### PR DESCRIPTION
**Pull request by Sandido**
## Description

This update adds ConfidentialVM as a securityType option to the Set-AzVmssSecurityProfile cmdlet and the accompanying SecurityEncryptionType parameter to the Set-AzVmssStorageProfile cmdlet. 
To support both values of the SecurityEncryptionType, I had to modify a previous feature in New-AzVmssConfig to support adding a gallery ImageReferenceId to the vmss, which was added to the New-AzVmss cmdlet previously. 

The testing setup for both SecurityEncryptionType parameters is rather complex. 

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs [hunk 1]
```diff
@@ -304,6 +304,12 @@ public partial class NewAzureRmVmssConfigCommand : Microsoft.Azure.Commands.Reso
             ValueFromPipelineByPropertyName = true)]
         public int RegularPriorityPercentage { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Specified the gallery image unique id for vmss deployment. This can be fetched from gallery image GET call.")]
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs:309]: message needs improvement

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssSecurityProfileCommand.cs [hunk 2]
```diff
@@ -70,6 +70,7 @@ private void Run()
                 }
                 this.VirtualMachineScaleSet.VirtualMachineProfile.SecurityProfile.SecurityType = this.SecurityType;
             }
+
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssSecurityProfileCommand.cs:73]: remove

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs [hunk 3]
```diff
@@ -162,6 +162,19 @@ public partial class SetAzureRmVmssStorageProfileCommand : Microsoft.Azure.Comma
         [PSArgumentCompleter("SCSI", "NVMe")]
         public string DiskControllerType { get; set; }
 
+        [Parameter(
+           Mandatory = false,
+           ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Sets the SecurityEncryptionType of the virtual machine scale set. possible values include: TrustedLaunch, ConfidentialVM_DiskEncryptedWithCustomerKey, ConfidentialVM_VMGuestStateOnlyEncryptedWithPlatformKey, ConfidentialVM_DiskEncryptedWithPlatformKey")]
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:168]: P ossible.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 4]
```diff
@@ -100,6 +100,23 @@ public override void ExecuteCmdlet()
                                 checkFlexibleOrchestrationModeParamsDefaultParamSet(parameters);
                             }
 
+
+                            if (this.IsParameterBound(c => c.ImageReferenceId))
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:104]: cleanup

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 5]
```diff
@@ -238,9 +255,6 @@ public override void ExecuteCmdlet()
                                 }
                             }
 
-
-
-
                             var psObject = new PSVirtualMachineScaleSet();
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:258]: Still need to dev for the other parameter sets.

Comment (Sandido) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:258]: No. These feature requests do not use that parameter set. Would be overextending the scope of this feature, and runs into many other issues as I have seen.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 4 above)
Comment (Sandido) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:104]: No. Will not pursue work in SimpleParamSet, non config.

**Comment by msJinLei** [comment]
```

  "Az.Compute","New-AzVmssConfig","1","3","Unassigned_Variable","5110","2","Set-AzVmssOsProfile -AdminPassword $AdminPassword is a null-valued parameter value.","-AdminPassword","Assign value for $AdminPassword."
  "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceOffer $ImgRef.Offer is a null-valued parameter value.","-ImageReferenceOffer","Assign value for $ImgRef.Offer."
  "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceSku $ImgRef.Skus is a null-valued parameter value.","-ImageReferenceSku","Assign value for $ImgRef.Skus."
  "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceVersion $ImgRef.Version is a null-valued parameter value.","-ImageReferenceVersion","Assign value for $ImgRef.Version."
  "Az.Compute","New-AzVmssConfig","1","6","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferencePublisher $ImgRef.PublisherName is a null-valued parameter value.","-ImageReferencePublisher","Assign value for $ImgRef.PublisherName."
  "Az.Compute","New-AzVmssConfig","1","6","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -VhdContainer $VHDContainer is a null-valued parameter value.","-VhdContainer","Assign value for $VHDContainer."
  "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -ComponentName $AUCComponentName is a null-valued parameter value.","-ComponentName","Assign value for $AUCComponentName."
  "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -Content $AUCContent is a null-valued parameter value.","-Content","Assign value for $AUCContent."
  "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -PassName $AUCPassName is a null-valued parameter value.","-PassName","Assign value for $AUCPassName."
  "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -SettingName $AUCSetting is a null-valued parameter value.","-SettingName","Assign value for $AUCSetting."
  "Az.Compute","New-AzVmssConfig","1","9","Unassigned_Variable","5110","2","New-AzVmss -ResourceGroupName $RGName is a null-valued parameter value.","-ResourceGroupName","Assign value for $RGName."
  "Az.Compute","New-AzVmssConfig","1","9","Unassigned_Variable","5110","2","New-AzVmss -Name $VMSSName is a null-valued parameter value.","-Name","Assign value for $VMSSName."
  "Az.Compute","Set-AzVmssSecurityProfile","2","40","Mismatched_Parameter_Value_Type","5111","2","Set-AzVmssOSProfile -AdminPassword $adminPassword is not an expected parameter value type.","-AdminPassword","Use correct parameter value type. Expected Type is string. Now the type is securestring.(Command)."
  "Az.Compute","Set-AzVmssStorageProfile","1","1","Mismatched_Parameter_Value_Type","5111","2","Set-AzVmssStorageProfile -VirtualMachineScaleSet 'ContosoVMSS' is not an expected parameter value type.","-VirtualMachineScaleSet","Use correct parameter value type. Expected Type is Microsoft.Azure.Commands.Compute.Automation.Models.PSVirtualMachineScaleSet. Now the type is string.(Static)."
  "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceOffer $ImgRef.Offer is a null-valued parameter value.","-ImageReferenceOffer","Assign value for $ImgRef.Offer."
  "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceSku $ImgRef.Skus is a null-valued parameter value.","-ImageReferenceSku","Assign value for $ImgRef.Skus."
  "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceVersion $ImgRef.Version is a null-valued parameter value.","-ImageReferenceVersion","Assign value for $ImgRef.Version."
  "Az.Compute","Set-AzVmssStorageProfile","1","3","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferencePublisher $ImgRef.PublisherName is a null-valued parameter value.","-ImageReferencePublisher","Assign value for $ImgRef.PublisherName."
  "Az.Compute","Set-AzVmssStorageProfile","1","3","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -VhdContainer $VhdContainer is a null-valued parameter value.","-VhdContainer","Assign value for $VhdContainer."
```

File: src/Compute/Compute/help/Set-AzVmssSecurityProfile.md [hunk 6]
```diff
@@ -30,7 +30,363 @@ $VMSS = Set-AzVmssSecurityProfile -VirtualMachineScaleSet $VMSS -SecurityType "T
 
 The first command gets the virtual machine scale set named ContosoVM07 by using **Get-AzVmss**.
 The command stores it in the $VMSS variable.
-The second command sets the SecurityType enum to "TrustedLaunch"
+The second command sets the SecurityType enum to "TrustedLaunch".
+
+### Example 2: Create a Confidential Vmss resource with encryption type VMGuestStateOnly.
```
Comment (msJinLei) [inline src/Compute/Compute/help/Set-AzVmssSecurityProfile.md:35]: Please revise your examples according to the static analysis report

```
2022-12-23T21:32:20.6208756Z   D:\a\_work\1\s\artifacts//StaticAnalysisResults\ExampleIssues.csv Errors
2022-12-23T21:32:20.6213126Z   "Module","Cmdlet","Example","Line","RuleName","ProblemId","Severity","Description","Extent","Remediation"
2022-12-23T21:32:20.6215066Z   "Az.Compute","Set-AzVmssSecurityProfile","2","36","Capitalization_Conventions_Violated","5101","1","New-AzVmssIPConfig doesn't follow the Capitalization Conventions.","New-AzVmssIPConfig -Name $vmssIPName -SubnetId $subnetId","Check the Capitalization Conventions. Suggest format: New-AzVmssIpConfig"
2022-12-23T21:32:20.6217431Z   "Az.Compute","Set-AzVmssSecurityProfile","2","40","Capitalization_Conventions_Violated","5101","1","Set-AzVmssOSProfile doesn't follow the Capitalization Conventions.","Set-AzVmssOSProfile -ComputerNamePrefix $computerNamePrefix -AdminUsername $adminUsername -AdminPassword $adminPassword","Check the Capitalization Conventions. Suggest format: Set-AzVmssOsProfile"
2022-12-23T21:32:20.6219812Z   "Az.Compute","Set-AzVmssSecurityProfile","3","50","Capitalization_Conventions_Violated","5101","1","Set-AzVmSecurityProfile doesn't follow the Capitalization Conventions.","Set-AzVmSecurityProfile -VM $VirtualMachine -SecurityType $vmSecurityType","Check the Capitalization Conventions. Suggest format: Set-AzVMSecurityProfile"
2022-12-23T21:32:20.6222034Z   "Az.Compute","Set-AzVmssSecurityProfile","3","51","Capitalization_Conventions_Violated","5101","1","Set-AzVmUefi doesn't follow the Capitalization Conventions.","Set-AzVmUefi -VM $VirtualMachine -EnableVtpm $true -EnableSecureBoot $true","Check the Capitalization Conventions. Suggest format: Set-AzVMUefi"
2022-12-23T21:32:20.6224181Z   "Az.Compute","Set-AzVmssSecurityProfile","3","109","Capitalization_Conventions_Violated","5101","1","New-AzVmssIPConfig doesn't follow the Capitalization Conventions.","New-AzVmssIPConfig -Name $vmssIPName -SubnetId $subnetId","Check the Capitalization Conventions. Suggest format: New-AzVmssIpConfig"
2022-12-23T21:32:20.6227060Z   "Az.Compute","Set-AzVmssSecurityProfile","3","128","Capitalization_Conventions_Violated","5101","1","Get-AzVmssvm doesn't follow the Capitalization Conventions.","Get-AzVmssvm -ResourceGroupName $rgname -VMScaleSetName $vmssName","Check the Capitalization Conventions. Suggest format: Get-AzVmssVM"
2022-12-23T21:32:20.6229209Z   "Az.Compute","Set-AzVmssSecurityProfile","3","129","Capitalization_Conventions_Violated","5101","1","Get-AzVmssvm doesn't follow the Capitalization Conventions.","Get-AzVmssvm -ResourceGroupName $rgname -VMScaleSetName $vmssName -InstanceId $vmssvms[0].InstanceId","Check the Capitalization Conventions. Suggest format: Get-AzVmssVM"
2022-12-23T21:32:20.6231714Z   "Az.Compute","Set-AzVmssSecurityProfile","4","15","Invalid_Cmdlet","5000","1","Get-PasswordForVM is not a valid command name.","Get-PasswordForVM","Check the spell of Get-PasswordForVM."
2022-12-23T21:32:20.6233265Z   "Az.Compute","Set-AzVmssSecurityProfile","4","17","Invalid_Cmdlet","5000","1","Get-ComputeTestResourceName is not a valid command name.","Get-ComputeTestResourceName","Check the spell of Get-ComputeTestResourceName."
2022-12-23T21:32:20.6235239Z   "Az.Compute","Set-AzVmssSecurityProfile","4","79","Capitalization_Conventions_Violated","5101","1","Set-AzVmSecurityProfile doesn't follow the Capitalization Conventions.","Set-AzVmSecurityProfile -VM $VirtualMachine -SecurityType $vmSecurityType","Check the Capitalization Conventions. Suggest format: Set-AzVMSecurityProfile"
2022-12-23T21:32:20.6237517Z   "Az.Compute","Set-AzVmssSecurityProfile","4","80","Capitalization_Conventions_Violated","5101","1","Set-AzVmUefi doesn't follow the Capitalization Conventions.","Set-AzVmUefi -VM $VirtualMachine -EnableVtpm $true -EnableSecureBoot $true","Check the Capitalization Conventions. Suggest format: Set-AzVMUefi"
2022-12-23T21:32:20.6239585Z   "Az.Compute","Set-AzVmssSecurityProfile","4","137","Capitalization_Conventions_Violated","5101","1","New-AzVmssIPConfig doesn't follow the Capitalization Conventions.","New-AzVmssIPConfig -Name $vmssIPName -SubnetId $subnetId","Check the Capitalization Conventions. Suggest format: New-AzVmssIpConfig"
2022-12-23T21:32:20.6241858Z   "Az.Compute","Set-AzVmssSecurityProfile","4","155","Capitalization_Conventions_Violated","5101","1","Get-AzVmssvm doesn't follow the Capitalization Conventions.","Get-AzVmssvm -ResourceGroupName $rgname -VMScaleSetName $vmssName","Check the Capitalization Conventions. Suggest format: Get-AzVmssVM"
2022-12-23T21:32:20.6245305Z   "Az.Compute","Set-AzVmssSecurityProfile","4","156","Capitalization_Conventions_Violated","5101","1","Get-AzVmssvm doesn't follow the Capitalization Conventions.","Get-AzVmssvm -ResourceGroupName $rgname -VMScaleSetName $vmssName -InstanceId $vmssvms[0].InstanceId","Check the Capitalization Conventions. Suggest format: Get-AzVmssVM"
2022-12-23T21:32:20.6246761Z   
2022-12-23T21:32:20.6247507Z   Following are warning issues. It is recommended to correct them as well.
2022-12-23T21:32:20.6248656Z   "Az.Compute","New-AzVmssConfig","1","1","Unassigned_Variable","5110","2","New-AzVmssConfig -Location $Loc is a null-valued parameter value.","-Location","Assign value for $Loc."
2022-12-23T21:32:20.6250140Z   "Az.Compute","New-AzVmssConfig","1","1","Unassigned_Variable","5110","2","New-AzVmssConfig -NetworkInterfaceConfiguration $NetCfg is a null-valued parameter value.","-NetworkInterfaceConfiguration","Assign value for $NetCfg."
2022-12-23T21:32:20.6251818Z   "Az.Compute","New-AzVmssConfig","1","2","Unassigned_Variable","5110","2","Add-AzVmssNetworkInterfaceConfiguration -IPConfiguration $IPCfg is a null-valued parameter value.","-IPConfiguration","Assign value for $IPCfg."
2022-12-23T21:32:20.6253265Z   "Az.Compute","New-AzVmssConfig","1","3","Unassigned_Variable","5110","2","Set-AzVmssOsProfile -AdminUsername $adminUsername is a null-valued parameter value.","-AdminUsername","Assign value for $adminUsername."
2022-12-23T21:32:20.6254411Z   "Az.Compute","New-AzVmssConfig","1","3","Unassigned_Variable","5110","2","Set-AzVmssOsProfile -AdminPassword $AdminPassword is a null-valued parameter value.","-AdminPassword","Assign value for $AdminPassword."
2022-12-23T21:32:20.6255267Z   "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceOffer $ImgRef.Offer is a null-valued parameter value.","-ImageReferenceOffer","Assign value for $ImgRef.Offer."
2022-12-23T21:32:20.6256671Z   "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceSku $ImgRef.Skus is a null-valued parameter value.","-ImageReferenceSku","Assign value for $ImgRef.Skus."
2022-12-23T21:32:20.6257552Z   "Az.Compute","New-AzVmssConfig","1","5","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceVersion $ImgRef.Version is a null-valued parameter value.","-ImageReferenceVersion","Assign value for $ImgRef.Version."
2022-12-23T21:32:20.6258464Z   "Az.Compute","New-AzVmssConfig","1","6","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferencePublisher $ImgRef.PublisherName is a null-valued parameter value.","-ImageReferencePublisher","Assign value for $ImgRef.PublisherName."
2022-12-23T21:32:20.6259344Z   "Az.Compute","New-AzVmssConfig","1","6","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -VhdContainer $VHDContainer is a null-valued parameter value.","-VhdContainer","Assign value for $VHDContainer."
2022-12-23T21:32:20.6260186Z   "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -ComponentName $AUCComponentName is a null-valued parameter value.","-ComponentName","Assign value for $AUCComponentName."
2022-12-23T21:32:20.6260997Z   "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -Content $AUCContent is a null-valued parameter value.","-Content","Assign value for $AUCContent."
2022-12-23T21:32:20.6261791Z   "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -PassName $AUCPassName is a null-valued parameter value.","-PassName","Assign value for $AUCPassName."
2022-12-23T21:32:20.6262706Z   "Az.Compute","New-AzVmssConfig","1","7","Unassigned_Variable","5110","2","Add-AzVmssAdditionalUnattendContent -SettingName $AUCSetting is a null-valued parameter value.","-SettingName","Assign value for $AUCSetting."
2022-12-23T21:32:20.6263499Z   "Az.Compute","New-AzVmssConfig","1","9","Unassigned_Variable","5110","2","New-AzVmss -ResourceGroupName $RGName is a null-valued parameter value.","-ResourceGroupName","Assign value for $RGName."
2022-12-23T21:32:20.6264201Z   "Az.Compute","New-AzVmssConfig","1","9","Unassigned_Variable","5110","2","New-AzVmss -Name $VMSSName is a null-valued parameter value.","-Name","Assign value for $VMSSName."
2022-12-23T21:32:20.6265088Z   "Az.Compute","Set-AzVmssSecurityProfile","2","40","Mismatched_Parameter_Value_Type","5111","2","Set-AzVmssOSProfile -AdminPassword $adminPassword is not an expected parameter value type.","-AdminPassword","Use correct parameter value type. Expected Type is string. Now the type is securestring.(Command)."
2022-12-23T21:32:20.6266306Z   "Az.Compute","Set-AzVmssStorageProfile","1","1","Mismatched_Parameter_Value_Type","5111","2","Set-AzVmssStorageProfile -VirtualMachineScaleSet 'ContosoVMSS' is not an expected parameter value type.","-VirtualMachineScaleSet","Use correct parameter value type. Expected Type is Microsoft.Azure.Commands.Compute.Automation.Models.PSVirtualMachineScaleSet. Now the type is string.(Static)."
2022-12-23T21:32:20.6267427Z   "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceOffer $ImgRef.Offer is a null-valued parameter value.","-ImageReferenceOffer","Assign value for $ImgRef.Offer."
2022-12-23T21:32:20.6268299Z   "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceSku $ImgRef.Skus is a null-valued parameter value.","-ImageReferenceSku","Assign value for $ImgRef.Skus."
2022-12-23T21:32:20.6269178Z   "Az.Compute","Set-AzVmssStorageProfile","1","2","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferenceVersion $ImgRef.Version is a null-valued parameter value.","-ImageReferenceVersion","Assign value for $ImgRef.Version."
2022-12-23T21:32:20.6270104Z   "Az.Compute","Set-AzVmssStorageProfile","1","3","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -ImageReferencePublisher $ImgRef.PublisherName is a null-valued parameter value.","-ImageReferencePublisher","Assign value for $ImgRef.PublisherName."
2022-12-23T21:32:20.6271049Z   "Az.Compute","Set-AzVmssStorageProfile","1","3","Unassigned_Variable","5110","2","Set-AzVmssStorageProfile -VhdContainer $VhdContainer is a null-valued parameter value.","-VhdContainer","Assign value for $VhdContainer."
```

**Review by msJinLei — DISMISSED** [review]

File: src/Compute/Compute/help/Set-AzVmssSecurityProfile.md (hunk 6 above)
Comment (Sandido) [reply src/Compute/Compute/help/Set-AzVmssSecurityProfile.md:35]: @msJinLei , resolved these issues.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**